core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -218,7 +226,7 @@ private ProjectNode createNewProjection(
             AggregationNode aggregationNode,
             ProjectNode projectNode,
             Map<CaseAggregation, Symbol> newProjectionSymbols,
-            Map<PreAggregationKey, PreAggregation> preAggregations)
+            Map<CaseAggregation, PreAggregation> lookupMap)
```

Comment:
Why map changed? Seems odd that we need to lookup by `CaseAggregation` here while `createPreProjection` and `createNewProjection` still use `PreAggregationKey`.

Did we loose some generalisation this way?

Reply:
> What's different with this change is that the keys of preAggregations can now have merged projections. That means we can no longer perform lookup by using a PreAggregationKey based on one case aggregation. It might not exist because it's part of the merged projections.

Could you give an example how it works?

Reply:
Sure, using the below test case
```
create table t (c1 varchar, c2 int);

insert into t values ('1', 1);
insert into t values ('2', 2);
insert into t values ('3', 3);
insert into t values ('4', 0);

select
  max(case when c1 = '1' then 10 / c2 end) as m1
, max(case when c1 = '2' then 10 / c2 end) as m2
, max(case when c1 = '3' then c2 end) as m3
, max(case when c1 = '4' then c2 end) as m4
from t;
```
After `createPreAggregations()` we will get
```
preAggregations:
<max, IF(c1='1' OR c1='2', 10/c2)> -> preagg(IF(c1='1' OR c1='2', 10/c2))
<max, c2> -> preagg(c2)

caseToPreAggregations:
m1 -> preagg(IF(c1='1' OR c1='2', 10/c2))
m2 -> preagg(IF(c1='1' OR c1='2', 10/c2))
m3 -> preagg(c2)
m4 -> preagg(c2)
```
As you can see, it's no longer possible for some cases to fetch the corresponding preaggregation from the map by using a `PreAggregationKey` built with the **original** case expression, because the updated `PreAggregationKey`s don't have 1:1 mapping with them.